### ISSUE TITLE
TSServer completions request: Cannot read property 'forEach' of undefined

### ISSUE THREAD

**Issue by mattbierner** [Microsoft/TypeScript#16610]
**TypeScript Version:**  2.4.1-insiders20170615

While working in a TS file with VS code, I see this error from the TSServer:

```
[Trace  - 12:37:05 PM] Response received: completions (4494). Request took 20 ms. Success: false . Message: Error processing request. Cannot read property 'forEach' of undefined
TypeError: Cannot read property 'forEach' of undefined
    at symbolsToArray (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:27957:20)
    at Object.getExportsOfModuleAsArray [as getExportsOfModule] (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:24310:20)
    at symbolCanbeReferencedAtTypeLocation (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:62880:55)
    at /Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:62844:36
    at Object.filter (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:1439:25)
    at filterGlobalCompletion (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:62830:27)
    at tryGetGlobalSymbols (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:62826:27)
    at getCompletionData (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:62718:22)
    at Object.getCompletionsAtPosition (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:62338:34)
    at Object.getCompletionsAtPosition (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:74358:35)
```

Not sure how to reliable reproduce it yet

**Comment by DanielRosenwasser** [comment]
Duplicate of #16610.

**Comment by mjbvz** [comment]
@DanielRosenwasser That's this bug. Reopening as a still see this in `@next`

**Comment by roblourens** [comment]
I repro this with 2.4.1 in VS Code when I open a file which contains only:

```ts
let x: 
```

And trigger completions at the end of the line.

TS server log:
[tsserver.txt](https://github.com/Microsoft/TypeScript/files/1118335/tsserver.txt)

Snippet:
```
Info 41   request: {"seq":10,"type":"request","command":"completions","arguments":{"file":"/Users/roblou/tmp/tsc/foo.ts","line":1,"offset":8}}
Err 42    Exception on executing command {"seq":10,"type":"request","command":"completions","arguments":{"file":"/Users/roblou/tmp/tsc/foo.ts","line":1,"offset":8}}:
Cannot read property 'forEach' of undefined
TypeError: Cannot read property 'forEach' of undefined
    at symbolsToArray (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:27963:20)
    at Object.getExportsOfModuleAsArray [as getExportsOfModule] (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:24316:20)
    at symbolCanbeReferencedAtTypeLocation (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:62889:55)
    at /Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:62853:36
    at Object.filter (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:1439:25)
    at filterGlobalCompletion (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:62839:27)
    at tryGetGlobalSymbols (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:62835:27)
    at getCompletionData (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:62727:22)
    at Object.getCompletionsAtPosition (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:62347:34)
    at Object.getCompletionsAtPosition (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:74367:35)
```

**Comment by roblourens** [comment]
Probably duplicate of #16537, maybe that's what you meant

**Comment by mjbvz** [comment]
@rbuckton Can we get this bug fixed for TS 2.4.2? I've seen it hit it in a number of cases and it now that @roblourens has provided a more reliable repo, we should be able to track down what is going wrong

**Comment by rbuckton** [comment]
@andy-ms Can you take a look at this? It may be related to [your comment](https://github.com/Microsoft/TypeScript/issues/16537#issuecomment-311690454) on #16537

**Comment by andy-ms** [comment]
Fixed by #17013

**Comment by mjbvz** [comment]
Thanks @andy-ms! Can we take this change into TS 2.4.2 as well?
